src/node.rs

```diff
@@ -301,36 +328,414 @@ impl Taffy {
     }
 }
 
-/// Internal node id.
-pub(crate) type NodeId = usize;
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::style::{Dimension, Display, FlexDirection};
 
-/// The identifier of a [`Node`]
-#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
-#[cfg_attr(not(any(feature = "std", feature = "alloc")), derive(hash32_derive::Hash32))]
-pub(crate) struct Id(usize);
+    /// New should be an empty with allocated size of 16
+    #[test]
+    fn test_new_is_default() {
+        // FIXME: This does not work unless we implement `PartialEq`, `Eq` on most types in the
+        // Is this a useful test to keep?
+        // assert_eq!(Taffy::new(), Taffy::default());
+    }
 
-/// An bump-allocator index that tracks how many [`Nodes`](Node) have been allocated in a [`Taffy`].
-pub(crate) struct Allocator {
-    /// The last reserved [`NodeId`]
-    last_id: AtomicUsize,
-}
+    #[test]
+    fn new_should_allocate_default_capacity() {
+        const DEFAULT_CAPACITY: usize = 16;
+        let taffy = Taffy::new();
+
+        assert!(taffy.nodes_to_ids.capacity() >= DEFAULT_CAPACITY);
+        assert!(taffy.ids_to_nodes.capacity() >= DEFAULT_CAPACITY);
+        assert!(taffy.forest.children.capacity() >= DEFAULT_CAPACITY);
+        assert!(taffy.forest.parents.capacity() >= DEFAULT_CAPACITY);
+        assert!(taffy.forest.nodes.capacity() >= DEFAULT_CAPACITY);
+    }
 
-impl Allocator {
-    /// Creates a fresh [`Allocator`]
-    #[must_use]
-    pub const fn new() -> Self {
-        Self { last_id: AtomicUsize::new(0) }
+    #[test]
+    fn test_with_capacity() {
+        const CAPACITY: usize = 8;
+        let taffy = Taffy::with_capacity(CAPACITY);
+
+        assert!(taffy.nodes_to_ids.capacity() >= CAPACITY);
+        assert!(taffy.ids_to_nodes.capacity() >= CAPACITY);
+        assert!(taffy.forest.children.capacity() >= CAPACITY);
+        assert!(taffy.forest.parents.capacity() >= CAPACITY);
+        assert!(taffy.forest.nodes.capacity() >= CAPACITY);
     }
 
-    /// Allocates space for one more [`Node`]
-    pub fn allocate(&self) -> Id {
-        Id(self.last_id.fetch_add(1, Ordering::Relaxed))
+    // #[test]
+    // fn allocate_node() {
+    //     let mut taffy = Taffy::new();
+    //     // let old_last_id = taffy.allocator.last_id.;
+    //     let id = taffy.allocate_node();
+
+    //     todo!("How to verify that this is working?");
+    // }
+
+    #[test]
+    fn add_node() {
+        let mut taffy = Taffy::new();
+        let node = taffy.allocate_node();
+        let id = taffy.forest.new_with_children(FlexboxLayout::default(), (&[]).to_vec());
+
+        // Should not contain nodes
+        assert!(!taffy.nodes_to_ids.contains_key(&node));
```

Comment:
Do you know why `contains_key` is private with [Features: None]? That looks like a bug to me.

Reply:
I have no idea :/ Maybe @alice-i-cecile have some insight?

Reply:
Ah, hang on, `IndexMap.contains_key()` needs to be `pub` for this to work I think.

Reply:
This was solved by exposing the methods on `IndexMap`